xarray/tests/test_merge.py

```diff
@@ -235,6 +235,264 @@ def test_merge_attrs_drop_conflicts(self):
         expected = xr.Dataset(attrs={"a": 0, "d": 0, "e": 0})
         assert_identical(actual, expected)
 
+    def test_merge_attrs_drop_conflicts_non_bool_eq(self):
```

Comment:
Both these test methods are very long, and could be split into 3 or 4 parts without any duplication. Can you please do that?